go/mysql/sql_error_test.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2020 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mysql
+
+import (
+	"testing"
+)
+
+func TestDumuxResourceExhaustedErrors(t *testing.T) {
+	type testCase struct {
+		msg  string
+		want int
+	}
+
+	cases := []testCase{
+		testCase{"misc", ERTooManyUserConnections},
+		testCase{"grpc: received message larger than max (99282+ vs. 1234): trailer", ERTooManyUserConnections},
+		testCase{"grpc: received message larger than max (1234 vs. 1234)", ERNetPacketTooLarge},
+		testCase{"header: grpc: received message larger than max (1234 vs. 1234)", ERNetPacketTooLarge},
+		testCase{"query payload size above threshold", ERNetPacketTooLarge},
+		testCase{"in-memory row count exceeded allowed limit of 13", ERNetPacketTooLarge},
+	}
+
+	for _, c := range cases {
+		got := demuxResourceExhaustedErrors(c.msg)
+		if got != c.want {
```

Comment:
nit: better to use `assert.Equal` for these comparisons

Reply:
✅